collector/nginx-ingresscontroller/README.md

```diff
@@ -0,0 +1,149 @@
+helm repo update
+helm install your-release-name -n your-collector-operator-namespace --create-namespace
+```
+
+3. Install the [NGINX Ingress Controller Operator](https://github.com/nginxinc/nginx-ingress-helm-operator#readme).
+
+The NGINX Ingress Controller Operator is a Helm based operator created with the [Operator Framework](https://sdk.operatorframework.io/). 
+
+Installing the NGINX Ingress Controller Operator requires some manual steps at this time. First we need a copy of the repo. Then we can use the Makefile in that repository root to complete the installation. We do that in this sequence of commands.
+
+```sh
+	git clone https://github.com/nginxinc/nginx-ingress-helm-operator/
+	cd nginx-ingress-helm-operator/
+	git checkout v1.0.0
+	make deploy IMG=nginx/nginx-ingress-operator:1.0.0
+```
+
+This action is in this repo's Makefile by the rule `install-nginx-ingress-operator`.
+
+5. Deploy an NGINX Ingress Controller instance
+
+The file at `ingress/values.yaml` tells the NGINX Ingress Controller Operator how to operate our instance. And `ingress/default-server-secret.yaml` provides a TLS cert for the purpose of the demo. 
+
+```sh
+kubectl apply -f ingress/
+```
+
+For purposes of monitoring via the Prometheus endpoint, we need to tend to two places in our configuration. In the first place we need to customPorts to our service. This needs to map the port where Prometheus metrics are exposed to a port where it will be exposed in the Kubernetes Service.
+
+```yaml
+    service:
+      create: true
+      type: LoadBalancer
+      customPorts:
+        - name: prometheus
+          port: 9113
+          targetPort: 9113
+```
+
+Now we need to enable Prometheus metrics. We do that in the Operator by configuring a prometheus section in the spec. The port we choose here needs to match what we map from in customPorts. And we need to be sure that create is true. We could configure security by adding a TLS cert as a secret for Prometheus. See the [NGINX docs](https://docs.nginx.com/nginx-ingress-controller/installation/installation-with-helm) for details of configuration with the helm chart.
+
+```yaml  
+  prometheus:
+    create: true
+    port: 9113
+    scheme: http
+```
+
+5. Deploy the Collector instance
+
+The first thing you need to do is configure the OTEL Collector's Prometheus Receiver to look for metrics at the endpoint where we've exposed by the NGINX Ingress Controller. You can approach this a variety of ways, but we're going to illustrate by embedding the configuration into the pod definition. In the spec you provide the Operator for the OpenTelemetryCollector kind you need to add a key for config, end the line with the | which indicates that a multi-line value follows.
+
+```yaml
+spec:
+  ...
+  config: |
+    receivers:
+      prometheus:
+        config:
+          scrape_configs:
+            - job_name: otel-nginx-eg
+              static_configs:
+                - targets: ["my-nginx-nginx-ingress:9113"]
```

Comment:
Does the above nginx example auto-create a service or endpoint for scraping? This also assumes it's in the same namespace as the collector which may not be the case. We should probably use the kube FQDN here.

Reply:
Addressed this in 1f48fddcd9be5473e10ea62904d82a98ee263f3b